src/main/webapp/js/datahandlers/collegelistdatahandler.js

```diff
@@ -19,28 +19,56 @@
 
 goog.module('datahandlers.collegelistdatahandler');
 
-const COLLEGE_LIST_ENDPT = '';
+const {CollegeQueryBuilder} = goog.require('datahandlers.collegequerybuilder');
+const {NAME, ACCEPTANCE_RATE, ACT_SCORE, ID} = 
+  goog.require('datahandlers.collegequerybuilder');
 
 /**
  * The data controller which fetches college data 
  * from backend and reformats the data. 
  */
 class CollegeListDataHandler {
 
-  constructor() {}
+  constructor() {
+  }
 
-  /** @returns The total number of scholarship stored in backend. */
-  async getTotalNumber() {
-    
+  /**
+   * This converts a json string to a js object map.
+   * @param {*} json - The json representing a list of colleges.
+   * @return - The list of js object maps.
+   * @private
+   */
+  convertJsonToObject_(json) {
+    let results = json["results"];
+    return results.map(element => this.converter_(element));
   }
 
   /**
-   * Fetch a batch of college object from DoE API.
-   * @param {number} batchIndex The index of the batch to be fetched.
-   * @param {number} itemsPerBatch The number of objects requested.
+   * Convert json string for single college to js object map.
+   * @param {*} element - The json string.
+   * @return {*} - JS object map representing a single college.
    */
-  async getNextBatch(batchIndex, itemsPerBatch) {
+  converter_(element) {
+    return {
+        name : element[NAME],
+        acceptance : element[ACCEPTANCE_RATE].toString(),
+        act : element[ACT_SCORE].toString(),
+        id : element[ID].toString()
+    };
+  }
```

Comment:
The code in this part is outdated. I use a list to represent each object, not sure if my changes will be approved. you can keep this code for now.